libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5410,7 +5455,13 @@ void GCS_MAVLINK::initialise_message_intervals_for_stream(GCS_MAVLINK::streams i
         // found it!
         const uint16_t interval_ms = get_interval_for_stream(id);
         for (uint8_t j=0; j<entries.num_ap_message_ids; j++) {
-            set_ap_message_interval(entries.ap_message_ids[j], interval_ms);
+            // ignore all streamrates (and set to 0) on a high latency link
+            if (is_high_latency_link && entries.ap_message_ids[j] != MAVLINK_MSG_ID_HIGH_LATENCY2 ) {
+                set_ap_message_interval(entries.ap_message_ids[j], 0);
+            }
+            else {
+                set_ap_message_interval(entries.ap_message_ids[j], interval_ms);
+            }
```

Comment:
```suggestion
            if (is_high_latency_link && entries.ap_message_ids[j] != MAVLINK_MSG_ID_HIGH_LATENCY2 ) {
                set_ap_message_interval(entries.ap_message_ids[j], 0);
            } else {
                set_ap_message_interval(entries.ap_message_ids[j], interval_ms);
            }
```

Reply:
This stuff makes me a bit unhappy, as if a user really wants to send out some other really small message over their high-latency link they can't.

Maybe we could zero all of the parameters if `HIGH_LATENCY2` has a zero interval AND we're a low-latency link - and then we set `HIGH_LATENCY2` non-zero.

Same comments apply elsewhere...

Reply:
User can still use ``MAV_CMD_SET_MESSAGE_INTERVAL`` if they _really_ want a message to come through

Reply:
Now removed this part